File: TT/TTP feitos/trabalho_teorico_2/palindromo_em_c/palindromo.c
#include <stdio.h>
#include <string.h>

void ler(char palavra[1000])
{
   char caractere;
   int i = 0;
  
   do
   {
        caractere = getchar();
        palavra[i] = caractere;
        ++i;
   } while(caractere != '\n');
   palavra[i-1]= '\0';
}


int eFim(char palavra[]){

int resp = 0;
    if (palavra[0] == 'F' && palavra[1] == 'I' && palavra[2] == 'M' && strlen(palavra)==3)
    {
        resp = 1;
    }
    
    return resp;
}


/*  FUNCAO PARA CHECAR SE E PALINDROMO 
 *  @param (palavra a ser checada)
 *   
 *
*/
void isPalindrome(char str[])
{
    // comecar da esquerda e da direita
    int l = 0;
    int h = strlen(str) - 1;
 
    // comparar
    while (h > l)
    {
        if (str[l++] != str[h--])
        {
            printf("NAO\n");
            return;
        }
    }
    printf("SIM\n");
}
 
//main
int main()
{

    char palavra[1000];
    
    ler(palavra);
    
    while (eFim(palavra) == 0)
    {
        isPalindrome(palavra);
        ler(palavra);
       
    }
    
    
    
    /*
    
    isPalindrome("abba");
    isPalindrome("abbccbba");
    isPalindrome("ana");
    isPalindrome("rafael");
    isPalindrome("a a");
    isPalindrome("a'a");
    isPalindrome("@!@");
    */
    
    return 0;
}

File: TT/TTP feitos/trabalho_teorico_2/palindromo_em_java/palindromo.java
class palindromo {

    public static boolean isFim(String s) {
        return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    public static String inverterPalavra(String palavra) {

        //int resp = 0;

        String reverso = "";

        for (int i = palavra.length() - 1; i >= 0; i--) {

            reverso += palavra.charAt(i);
            // MyIO.println(reverso);

        }

        return reverso;
    }

    public static boolean compararPalavras(String palavra, String reverso) {

        boolean resp = true;
        for (int i = 0; i < palavra.length(); i++) {

            if (palavra.charAt(i) != reverso.charAt(i)) {
                resp = false;

            }
        }

        return resp;
    }

    public static void main(String[] args) {

        String palavra = "";
        String reverso = "";

        palavra = MyIO.readLine();
        while(isFim(palavra)==false){
            reverso = inverterPalavra(palavra);
            
            if (compararPalavras(palavra, reverso)) 
            {
                MyIO.println("SIM");
            
            } else {
            
                MyIO.println("NAO");
            
            }

            palavra = MyIO.readLine();
        }

    }

}